mvx-esdt-safe/src/lib.rs

```diff
@@ -24,11 +24,22 @@ pub trait MvxEsdtSafe:
     fn init(
         &self,
         header_verifier_address: ManagedAddress,
-        esdt_safe_config: EsdtSafeConfig<Self::Api>,
+        opt_config: OptionalValue<EsdtSafeConfig<Self::Api>>,
     ) {
         self.require_sc_address(&header_verifier_address);
         self.header_verifier_address().set(&header_verifier_address);
-        self.esdt_safe_config().set(esdt_safe_config);
+
+        self.esdt_safe_config().set(
+            opt_config
+                .into_option()
+                .unwrap_or_else(EsdtSafeConfig::default_config),
+        );
+    }
+
+    #[only_owner]
+    #[endpoint(updateConfiguration)]
+    fn update_configuration(&self, new_config: EsdtSafeConfig<Self::Api>) {
+        self.esdt_safe_config().set(new_config);
```

Comment:
Create a task for later to check that `new_config` is valid
1. max tx gas limit should be maximum `erd_max_gas_per_transaction` from [here](https://gateway.multiversx.com/network/config) - if you don't have access to this, hard-code the value